### ISSUE TITLE
wp core download "does not seem to be a WordPress install"

### ISSUE THREAD

**Issue by jaggedsoft** [wp-cli/wp-cli#2037]
wp core download --force
Now that I have upgraded wp-cli (My previous version was just before you added the --allow-root flag) I can no longer run the "wp core download" command.
I have an empty directory I've navigated to, and its' permissions are +rwx 0755.

wp core download --force
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`.

You can imagine the frustration from the command `wp core download` telling you to run `wp core download`

**Comment by danielbachhuber** [comment]
Not sure I follow what the bug report is. `wp core download --force` works fine for me in an empty directory:

```
salty-wordpress ➜  wp-test.dev  ls -al
total 0
drwxr-xr-x 1 vagrant vagrant   68 Sep  7 14:41 .
drwxr-xr-x 1 vagrant vagrant 1088 Sep  7 14:41 ..
salty-wordpress ➜  wp-test.dev  wp core download --force
Downloading WordPress 4.3 (en_US)...
Success: WordPress downloaded.
```

Can you clarify?

**Comment by jaggedsoft** [comment]
I am in fact trying to install wordpress into an empty directory. I'm currently switched to that directory and I run 'wp core update'. This command has worked for months and I'm not sure why it's doing this since I updated.
--debug is also not helping to explain the situation. No different:
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`. (I've tried --path, no change.)
My issue seems to happen due to core.php's version(). It's looking for wp-includes/version.php and it doesn't exist yet. The folder and all before it are permissions 0755.

**Comment by danielbachhuber** [comment]
> I'm currently switched to that directory and I run 'wp core update'.

Are you running `wp core download` or `wp core update` ? They're two separate commands, you know.

Also, can you share the directory structure?

**Comment by jaggedsoft** [comment]
It's `wp core download` ..my mistake.
/home/jaggedsoft/nextlocal/nextlocal.ca
/home/jaggedsoft/nextlocal/nextlocal.me
/home/jaggedsoft/nextlocal/nextlocal.io
/home/jaggedsoft/nextlocal/nextlocal.us
/home/jaggedsoft/nextlocal/nextlocal.net
/home/jaggedsoft/nextlocal/nextlocal.solar
/home/jaggedsoft/nextlocal/staffing-agency.io
/home/jaggedsoft/nextlocal/contractor.best
/home/jaggedsoft/nextlocal/electrician.best
/home/jaggedsoft/nextlocal/garage-door.io
/home/jaggedsoft/nextlocal/mortgage-agent.io
/home/jaggedsoft/nextlocal/mortgage-broker.io
/home/jaggedsoft/nextlocal/locksmith.best
/home/jaggedsoft/nextlocal/plumber.best
/home/jaggedsoft/nextlocal/pest-control.best
And I keep all my customer hosted sites in /home/jaggedsoft/nextlocal/hosted/
In this case I wanted to create a new domain, /home/jaggedsoft/nextlocal/nextlocal.solar
I navigate to the new folder I just created, and try `wp core download` and the result is this:
jaggedsoft@nextlocal:~/nextlocal/nextlocal.solar# wp core download
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`.

I have since created my own shell script to download and install Wordpress, I just thought I'd let you know that myself and some of my friends have started running into this bizarre issue and I can not explain it.

**Comment by jaggedsoft** [comment]
The directory structure has correct ownership/group and permissions all the way down. I've also tried as root:
root@nextlocal:~/nextlocal/nextlocal.solar# wp core download --allow-root --force --debug
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`.
root@nextlocal:~/nextlocal/nextlocal.solar#

There is nothing else in the directory. Thanks for looking in to this. If you need me to test anything, it's not a problem.

**Comment by danielbachhuber** [comment]
Sorry for the late reply. Were you able to sort this out?

If not, can you share `wp --info`? And, does `~/nextlocal/nextlocal.solar` have an `index.php` file in it, by chance?

**Comment by jaggedsoft** [comment]
Nope, newly created directory with correct permissions and ownership. System is 

    PHP binary:     /usr/bin/php5
    PHP version:    5.6.13-0+deb8u1
    php.ini used:   /etc/php5/cli/php.ini
    WP-CLI root dir:        phar://wp-cli.phar
    WP-CLI global config:
    WP-CLI project config:
    WP-CLI version: 0.20.0

Just for kicks, updated to WP-CLI version 0.20.1. System `Linux 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt11-1+deb8u3 (2015-08-04) x86_64 GNU/Linux`

Here's a new directory and I'm testing as root:
    root@nextlocal:/home/jaggedsoft/nextlocal/multisite/nextlocal.pw# ls -a
    .  .. (empty)
    root@nextlocal:/home/jaggedsoft/nextlocal/multisite/nextlocal.pw# wp core download --allow-root --force --debug
    Error: This does not seem to be a WordPress install.
    Pass --path=`path/to/wordpress` or run `wp core download`.
    root@nextlocal:/home/jaggedsoft/nextlocal/multisite/nextlocal.pw#

Thanks for trying to figure this out.

**Comment by danielbachhuber** [comment]
Not sure what this is, but it seems to be a problem specific to your environment. Sorry I can't help further.

**Comment by jaggedsoft** [comment]
I understand but I would still like if you could add more debugging information to help figure out why

**Comment by GavinCS** [comment]
I can confirm that I am having the same issue.
I am running  ```wp core download``` in an empty directory and receive the same error message

**Comment by prosti** [comment]
I can confirm the same running wp core download in an empty directory. I receive the same error message.
My wp --info
PHP binary:	/usr/bin/php5
PHP version:	5.6.11-1ubuntu3.1
php.ini used:	/etc/php5/cli/php.ini
WP-CLI root dir:	phar://wp-cli.phar
WP-CLI global config:	
WP-CLI project config:	
WP-CLI version:	0.22.0

**Comment by danielbachhuber** [comment]
> I can confirm the same running wp core download in an empty directory

Can you share `ls -al` ? Also, can you run `wp core download --debug` and share the full output?

**Comment by jaggedsoft** [comment]
If you would like an alternative to `wp core download`, this will create a new wordpress install in your current directory named 'wordpress'
## `wget https://wordpress.org/latest.tar.gz && tar -xzvf latest.tar.gz`

**Comment by jaggedsoft** [comment]
Just to confirm, this happens for me in a brand new, completely empty directory, whether I'm root or not. My permissions for every folder are 0775. (rwx rwx r-x) and all owned by the correct user.

**Comment by danielbachhuber** [comment]
> The output of `wp core download --debug` for me is:
> > Debug: ABSPATH defined: /home/jaggedsoft/nextlocal/hosted/nextlocal.website/ (0.037s)

`wp core download` is supposed to be hooked on to `before_wp_load` ([ref](https://github.com/wp-cli/wp-cli/blob/1b8da36b766bed230c736d15ded86cf5f4fe53a0/php/commands/core.php#L73)). If it's getting to "Error: This does not seem to be a WordPress install.", then WP-CLI isn't calling `core download on the `before_wp_load` hook ([ref](https://github.com/wp-cli/wp-cli/blob/1b8da36b766bed230c736d15ded86cf5f4fe53a0/php/WP_CLI/Runner.php#L673-L675)).

Are you able to reproduce in another environment, or in just this one? Did you install PHP in this environment?

> Could we please re-open this and drop the state:unconfirmed tag now that there's at least 3 people experiencing this?

1. Re-opening the issue doesn't do anything to solve the problem in your environment.
2. The issue is `state:unconfirmed` because it's not a confirmed issue — it only reproduces in maybe one other environment.

As a volunteer-maintained free and open source project, WP-CLI offers no guarantees of user support. The issue tracker is intended as a development backlog, and for confirmed bugs.

**Comment by jaggedsoft** [comment]
> Are you able to reproduce in another environment? it only reproduces in maybe one other environment.
Myself, prosti, GavinCS above all have the same issue. I cannot speak about them but I'm running on a pretty ordinary debian installation off of digitalocean. Of course PHP is installed.

On line 747 of core.php you will see:
`if ( !is_readable( $versions_path )`

The issue is you're triggering an error if you want to install in a blank folder, because is_readable is returning false since there is no "wp-includes/version.php"

Does that make sense?
/home/jaggedsoft/nextlocal/hosted/nextlocal.website/ exists, it's an empty folder. Therefore is_readable() on that folder is true.
However, is_readable() fails on /home/jaggedsoft/nextlocal/hosted/nextlocal.website/wp-includes/version.php because wp-includes is not a folder (yet)

`wp core download` should work on completely empty folders, should it not?

**Comment by danielbachhuber** [comment]
> Myself, prosti, GavinCS above all have the same issue.

You all have the same reported symptoms. This doesn't necessarily mean you have the same issue.

> Of course PHP is installed.

Sorry, I meant did _you_ install PHP in that environment? Do you know the specifics of the configuration, and whether there might be anything special about it?

> On line 747 of core.php you will see:
> The issue is you're triggering an error if you want to install in a blank folder, because is_readable is returning false since there is no "wp-includes/version.php"

Right, because that method is mapped to `wp core version`. If you run `wp core version` in an empty directory, you'll get that error, as I do:

```
salty-wordpress ➜  empty-dir  wp core version
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`.
```

> `wp core download` should work on completely empty folders, should it not?

It does in my testing, and for many other users:

```
salty-wordpress ➜  empty-dir  wp core download
Downloading WordPress 4.4.2 (en_US)...
Using cached file '/home/vagrant/.wp-cli/cache/core/wordpress-4.4.2-en_US.tar.gz'...
Success: WordPress downloaded.
```

**Comment by prosti** [comment]
Hello,
I my case I must to say I don't have this problem any more. I had a script and I think it was a typo...
like `wp core downlaod`. It was my bad. Sorry.

**Comment by wonder32** [comment]
for me wp cli threw this when I tried commands against a mounted network drive..
but it works when the wordpress setup is on the local machine

**Comment by Lewiscowles1986** [comment]
for me this was because 4.6 update failed part way through (unknown why as it did not error out), and that led to version.php missing. 

 * Updated my wp-cli
 * Downloaded the WP 4.6
 * unpacked everything apart from wp-content over my existing WP install
 * cleared opcache 'just-in-case'

Then everything worked. Like most things WordPress-y, determination and focusing on honing process have a lot to offer

**Comment by yuniordaga** [comment]
You must be on the path where your WordPress has the files as well as:
/var/www/html: 
            -here we have the files 
../var/www/html # wp plugin list

**Comment by malaney** [comment]
I ran into this issue and discovered that not all wp-cli's are created equal.  I was NOT able to run `wp core download` when I first installed wp-cli in my docker container using `composer create-project wp-cli/wp-cli-bundle --no-dev`.  However, when I installed wp-cli using the PHAR (i.e. `curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar`) I was able to run `wp core download` in an empty directory and get a fresh download of WP 6.1.1 with no errors ... (aside from the PHP 8.1 deprecation notices.  See https://github.com/wp-cli/wp-cli/issues/5623)
